### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by vkhanikyan**

### CODE REVIEW

File: src/main/java/com/ubs/vahan/test/AppMain.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class AppMain {
+    private static final Logger LOGGER = Logger.getLogger(AppMain.class.getName());
+
+    public static void main(String[] args) {
+        if (args.length == 0) {
+            System.out.println("Please give the path of the csv file as a parameter");
```
Comment (hovhannes-khalatyan) [inline src/main/java/com/ubs/vahan/test/AppMain.java:13]: if there is no any argument, maybe we should here also use System.exit(1);

File: src/main/java/com/ubs/vahan/test/CSVReaderSudoku.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Used enum as it is comfortable to use it as util class
+ */
+public enum CSVReaderSudoku {
+    ;
```
Comment (hovhannes-khalatyan) [inline src/main/java/com/ubs/vahan/test/CSVReaderSudoku.java:20]: you can remove this semicolon from here :)

File: src/main/java/com/ubs/vahan/test/CSVReaderSudoku.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Used enum as it is comfortable to use it as util class
```
Comment (hovhannes-khalatyan) [inline src/main/java/com/ubs/vahan/test/CSVReaderSudoku.java]: add more detailed description why we need enum here

File: src/main/java/com/ubs/vahan/test/AppMain.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+            return;
+        }
+        try {
+            String path = args[0];
+            int[][] board = CSVReaderSudoku.readInts(path);
+            boolean isValid = SudokuValidator.isValid(board);
+            System.out.println(isValid ? "it is valid Sudoku board" : "it is invalid Sudoku board");
+            System.exit(0);
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Exception occur", e);
```
Comment (hovhannes-khalatyan) [inline src/main/java/com/ubs/vahan/test/AppMain.java:31]: Add system out exception message here.

File: src/main/java/com/ubs/vahan/test/SudokuValidator.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        validateWidth(board);
+        validateValues(board);
+
+    }
+
+    private static void validateValues(int[][] board) {
+        for (int[] row : board) {
+            for (int item : row) {
+                if (!(item >= 0 && item <= 9)) {
+                    throw new IllegalArgumentException(String.format("The board item has value not in 1-9 range, it is: %d", item));
```
Comment (hovhannes-khalatyan) [inline src/main/java/com/ubs/vahan/test/SudokuValidator.java:58]: Will be great if your error message contains invalid data coordinates. e.g. You have a invalid data -1 in: row 3, col 5.

File: src/main/java/com/ubs/vahan/test/AppMain.java (hunk 1 above)
Comment (vkhanikyan) [reply src/main/java/com/ubs/vahan/test/AppMain.java:13]: Done

File: src/main/java/com/ubs/vahan/test/AppMain.java (hunk 4 above)
Comment (vkhanikyan) [reply src/main/java/com/ubs/vahan/test/AppMain.java:31]: Done

File: src/main/java/com/ubs/vahan/test/CSVReaderSudoku.java (hunk 3 above)
Comment (vkhanikyan) [reply src/main/java/com/ubs/vahan/test/CSVReaderSudoku.java]: Done

File: src/main/java/com/ubs/vahan/test/CSVReaderSudoku.java (hunk 2 above)
Comment (vkhanikyan) [reply src/main/java/com/ubs/vahan/test/CSVReaderSudoku.java:20]: As it is an enum, it needs this semicolon to compile. I'd be happy to do that, but can't do it with this structure :)

File: src/main/java/com/ubs/vahan/test/SudokuValidator.java (hunk 5 above)
Comment (vkhanikyan) [reply src/main/java/com/ubs/vahan/test/SudokuValidator.java:58]: Done in an intermediate way. Print the column, row, square values were found the problem

**Pull request merged**